src/api/block/controllers/block.ts

```diff
@@ -17,32 +17,51 @@ export default factories.createCoreController(
     async findByVuid(ctx) {
       try {
         const { vuid } = ctx.params
-        const { populate, locale } = ctx.query
+        const {
+          populate,
+          locale,
+          fallbackToDefaultLocale = 'false',
+        } = ctx.query
 
-        const entries: Block[] = await strapi.entityService.findMany(
-          'api::block.block',
-          {
+        let entries: Block[] = []
+
+        const requestedLocaleEntries: Block[] =
+          await strapi.entityService.findMany('api::block.block', {
             filters: {
               vuid,
             },
             locale: locale ?? 'en',
             populate,
-          }
-        )
+          })
+
+        entries = [...requestedLocaleEntries]
+
+        if (
+          fallbackToDefaultLocale === 'true' &&
+          (entries === undefined || entries.length === 0)
+        ) {
+          const defaultLocaleEntries: Block[] =
+            await strapi.entityService.findMany('api::block.block', {
+              filters: {
+                vuid,
+              },
+              locale: 'en',
+              populate,
+            })
 
-        if (entries === undefined || entries.length === 0) {
-          ctx.status = 404
-          ctx.body = `Unable to find block with vuid ${vuid}`
-          return
+          entries = [...defaultLocaleEntries]
```

Comment:
Same here, can `entries` be replaced instead of creating a new variable and immediately spread?